lib/ace/mode/lean_highlight_rules.js

```diff
@@ -0,0 +1,200 @@
+                                 "/\\", "\\/", "∧", "∨", "≠", "<", ">", "≤", "≥", "¬",
+                                 "<=", ">=", "⁻¹", "⬝", "▸", "\\+", "\\*", "-", "/",
+                                 "λ", "→", "∃", "∀", ":="].join("|"));
+    // regexp must not have capturing parentheses. Use (?:) instead.
+    // regexps are ordered -> the first match is used
+
+    this.$rules = {
+        "start" : [
+            {
+                token : "comment", // single line comment "--"
+                regex : "--.*$"
+            },
+            DocCommentHighlightRules.getStartRule("doc-start"),
+            {
+                token : "comment", // multi line comment "/-"
+                regex : "\\/-",
+                next : "comment"
+            }, {
+                token : "string", // single line
+                regex : '["](?:(?:\\\\.)|(?:[^"\\\\]))*?["]'
+            }, {
+                token : "string", // multi line string start
+                regex : '["].*\\\\$',
+                next : "qqstring"
+            }, {
+                token : "string", // single line
+                regex : "['](?:(?:\\\\.)|(?:[^'\\\\]))*?[']"
+            }, {
+                token : "string", // multi line string start
+                regex : "['].*\\\\$",
+                next : "qstring"
+            }, {
+                token : "constant.numeric", // hex
+                regex : "0[xX][0-9a-fA-F]+(L|l|UL|ul|u|U|F|f|ll|LL|ull|ULL)?\\b"
+            }, {
+                token : "constant.numeric", // float
+                regex : "[+-]?\\d+(?:(?:\\.\\d*)?(?:[eE][+-]?\\d+)?)?(L|l|UL|ul|u|U|F|f|ll|LL|ull|ULL)?\\b"
+            }, {
+                token : "storage.modifier",
+                regex : storageModifiers
+            }, {
+                token : "keyword", // pre-compiler directives
+                regex : "#\\s*(?:include|import|pragma|line|define|undef|if|ifdef|else|elif|ifndef)\\b",
+                next  : "directive"
+            }, {
+                token : "keyword", // special case pre-compiler directive
+                regex : "(?:#\\s*endif)\\b"
+            }, {
+                token : keywordMapper,
+                regex : identifierRe
+            }, {
+                token : "operator",
+                regex : operatorRe
+            }, {
+              token : "punctuation.operator",
+              regex : "\\?|\\:|\\,|\\;|\\."
+            }, {
+                token : "paren.lparen",
+                regex : "[[({]"
+            }, {
+                token : "paren.rparen",
+                regex : "[\\])}]"
+            }, {
+                token : "text",
+                regex : "\\s+"
+            }
+        ],
+        "comment" : [ {token: "comment", regex: "-/", next: "start"},
+                      {defaultToken: "comment"} ],
+        "qqstring" : [ {token : "string", regex : '(?:(?:\\\\.)|(?:[^"\\\\]))*?"', next : "start" },
```

Comment:
This should be something like

```js
[ {
     token : "constant.language.escape",
     regex : /\\./
}, {
     token : "string",
     regex : '"',
     next : "start" 
}, {
    defaultToken: "string"
}]
```

This depends on whether strings are multiline or not, and are there special escape patterns like \uXXXX